temporal-sdk/src/main/java/io/temporal/client/ListWorkflowExecutionIterator.java

```diff
@@ -11,14 +11,14 @@
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-class ListWorkflowExecutionIterator
+public class ListWorkflowExecutionIterator
```

Comment:
Does this need to be public to support the interceptor? In .NET for example we felt there was no need to expose the underlying machinery of list workflows because creating an async iterator (or Java stream) should be easy enough if they _must_ completely replace the logic and not just delegate to the next interceptor.

Reply:
ideally no, the problem is the [RootWorkflowClientInvoker.java](https://github.com/temporalio/sdk-java/pull/2524/files#diff-7209d2833e3d842e729546850af071d4522573f109e0c979cf21314be7488bba) is in internal so it can't be private, moving this class causes a lot of other issues

Reply:
> moving this class causes a lot of other issues

Hrmm. I wonder making public _and_ moving from `io.temporal.client` to `io.temporal.client.internal` would make sense.

Reply:
Let me try, I think I had some issue doing that since some other classes in `io.temporal.client` also have Iterators

Reply:
Moved this class and `EagerPaginator` to internal